### PR TITLE
made "concurernvy vs. parallelism" section clearer

### PR DESCRIPTION
**Pull request by alexbooker**
For many people, the concept of concurrency vs. parallelism can be hard to grok.

I do not think the old descriptin read very well, nor did I think it would be _that_ helpful to someone who needed the answer in the frirst place (i.e. someone who doesn't know the difference.)

I tried writing an answer of my own before I realized that someone else had already put it best, so I quoted an aclaimed StackOverflow answer and gave appropiate credit (as Creative Commons requires).

I think linking to a SO question will also make it easy for people to read other answers, which they may understand better.

### CODE REVIEW

**Comment by schnittstabil** [comment]
Nice, I like clarifications, but I don't think he *put it best*. Mainly because the examples explain more than the descriptions, but introduces (probably unknown) terms: *multitasking*, *single-core*, *multicore*, *processor*.

File: readme.md [hunk 1]
```diff
@@ -520,7 +520,13 @@ AVA, not Ava or ava. Pronounced [`/ˈeɪvə/` ay-və](media/pronunciation.m4a?ra
 
 ### Concurrency vs. parallelism
 
-Concurrency is not parallelism. It enables parallelism. It's about dealing with, while parallelism is about doing, lots of things at once.
+Concurrency is not parallelism. It enables parallelism.
+
+**Concurrency** is when two tasks can start, run, and complete in overlapping time periods. It doesn't necessarily mean they'll ever both be running at the same instant. (e.g. multitasking on a single-core machine.)
```
Comment (schnittstabil) [inline readme.md]: 1. *two or more tasks*?
2. I wouldn't say *starting* and *completing* is necessary
3. I'm not a native speaker, but I wouldn't use the word `when` in that context to express a condition.

**Comment by alexbooker** [comment]
Thanks for the input, @schnittstabil - I apprecaite it :smile:.

I _completely_ agree that it should be _"two ore more tasks"_ - I edited the original [answer](https://stackoverflow.com/posts/1050257/revisions) and this pull request in response.

Bare in mind the answer has **400:heavy_plus_sign:**   positive responses which tells me the answer _is_ useful.

I think it reads fine, to be honest. What's more, because I have included a link to the SO thread someone can go there for additional context and alternative answers or analogies.

File: readme.md [hunk 2]
```diff
@@ -520,7 +520,13 @@ AVA, not Ava or ava. Pronounced [`/ˈeɪvə/` ay-və](media/pronunciation.m4a?ra
 
 ### Concurrency vs. parallelism
 
-Concurrency is not parallelism. It enables parallelism. It's about dealing with, while parallelism is about doing, lots of things at once.
+Concurrency is not parallelism. It enables parallelism.
+
+**Concurrency** is when two or more tasks can start, run, and complete in overlapping time periods. It doesn't necessarily mean they'll ever both be running at the same instant. (e.g. multitasking on a single-core machine.)
```
Comment (schnittstabil) [inline readme.md]: *two or more tasks can start, run, and complete in overlapping time periods*, that is also true for **Parallelism**, isn't it?

Comment (alexbooker) [inline readme.md]: @schnittstabil As is stated: _"concurrency enables parallism"_. Of course there is an overlap.

What do you suggest?

**Comment by vdemedes** [comment]
I think this information should not belong in the AVA's Readme. Project's readme is about project, not about explaining concepts. What I would do here, is to link to that stack overflow answer, instead of copying it.

**Comment by alexbooker** [comment]
@vdemedes You know what, mate. That is a good idea. Done.

**Comment by schnittstabil** [comment]
I thought about that description and has come to the conclusion that it is at least misleading.
*Sequential* and *Concurrent* is about algorithms and language design.<br/>*Serial* and *Parallel* is about hardware and run-time.

Instead of covering these aspects they are messed up.

Also the answer mentions *Sun's* **Multithreaded** *Programming Guide*, which (obviously) only talking about Threads and states that: Concurrency is a more generalized form of parallelism. That is only valuable in that guide, but can't be transferred to newer concepts used in Node or Go…

**Comment by alexbooker** [comment]
@schnittstabil I meant to link to the question.

**Pull request closed without merging**